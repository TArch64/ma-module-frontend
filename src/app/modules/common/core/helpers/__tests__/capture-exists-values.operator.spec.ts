import {of} from "rxjs";
import {toArray} from "rxjs/operators";
import {captureExistsValues} from "../capture-exists-values.operator";

describe('capture exists values', () => {
    test('should capture exists value', (done) => {
        of(123, 'test', {}, []).pipe(captureExistsValues, toArray()).subscribe(values => {
            expect(values).toMatchSnapshot();
            done();
        });
    });

    test('should ignore null', (done) => {
        of(123, null).pipe(captureExistsValues, toArray()).subscribe(values => {
            expect(values).toMatchSnapshot();
            done();
        });
    });

    test('should ignore undefined', (done) => {
        of(123, undefined).pipe(captureExistsValues, toArray()).subscribe(values => {
            expect(values).toMatchSnapshot();
            done();
        });
    });

    test('should not ignore empty string', (done) => {
        of(123, '').pipe(captureExistsValues, toArray()).subscribe(values => {
            expect(values).toMatchSnapshot();
            done();
        });
    });

    test('should not ignore false', (done) => {
        of(123, false).pipe(captureExistsValues, toArray()).subscribe(values => {
            expect(values).toMatchSnapshot();
            done();
        });
    });
});
