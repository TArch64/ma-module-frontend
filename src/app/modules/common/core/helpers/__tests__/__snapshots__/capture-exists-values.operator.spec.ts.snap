// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`capture exists values should capture exists value 1`] = `
Array [
  123,
  "test",
  Object {},
  Array [],
]
`;

exports[`capture exists values should ignore null 1`] = `
Array [
  123,
]
`;

exports[`capture exists values should ignore undefined 1`] = `
Array [
  123,
]
`;

exports[`capture exists values should not ignore empty string 1`] = `
Array [
  123,
  "",
]
`;

exports[`capture exists values should not ignore false 1`] = `
Array [
  123,
  false,
]
`;
